lib/Pusher.php

```diff
@@ -258,8 +269,10 @@ private function validate_socket_id($socket_id)
     /**
      * Utility function used to create the curl object with common settings.
      */
-    private function create_curl($s_url, $request_method = 'GET', $query_params = array())
+    private function create_curl($s_url, $request_method = 'GET', $query_params = array(), $notification_host = false)
```

Comment:
I think we need a function that takes a base_url such as `$base_url . $s_url .'?'.$signed_query` gives the full url. Then introduce a `create_notification_curl` and `create_ddn_curl` wrapper functions that pass the right base_url.